internal/api/v1/auth.go

```diff
@@ -0,0 +1,68 @@
+package v1
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/epinio/epinio/internal/api/v1/response"
+	"github.com/epinio/epinio/internal/auth"
+	"github.com/epinio/epinio/internal/cli/server/requestctx"
+	apierrors "github.com/epinio/epinio/pkg/api/core/v1/errors"
+	"github.com/gin-gonic/gin"
+	"github.com/go-logr/logr"
+)
+
+func AuthorizationMiddleware(c *gin.Context) {
+	logger := requestctx.Logger(c.Request.Context()).WithName("AuthorizationMiddleware")
+	user := requestctx.User(c.Request.Context())
+
+	method := c.Request.Method
+	path := c.Request.URL.Path
+	namespace := c.Param("namespace")
+
+	logger.Info(fmt.Sprintf("authorization request from user [%s] with role [%s] for [%s - %s]", user.Username, user.Role, method, path))
+
+	var authorized bool
+	switch user.Role {
+	case "admin":
+		authorized = authorizeAdmin(logger)
+	case "user":
+		authorized = authorizeUser(logger, user, path, namespace)
+	}
+
+	logger.Info(fmt.Sprintf("user [%s] authorized [%t]", user.Username, authorized))
```

Comment:
Recommend to show role information, and maybe also namespace.

Reply:
I've logged the role in line 23, I will add also the namespaces, I don't think they will be many. :slightly_smiling_face:

Reply:
I referred here only to __the__ namespace, i.e. the targeted one. Not the list of allowed namespaces

Reply:
Ah, I see. It can be seen in the path then, but it's nice to have it also here for clarification!